python/tests/test_testclient_shared_connection.py

```
@@ -0,0 +1,299 @@
+
+@pytest.mark.django_db
+def test_a_second_thread_cannot_enter_its_own_client():
+    """The shared store is process-wide, thus a second test must not join it.
+
+    Two threads that each enter a client would lend the connection of one test
+    to the handlers of the other, which mixes their rows. No supported runner
+    runs two tests in one process at the same time, thus this is refused.
+    """
+    outcome: list[object] = []
+
+    def other_test():
```

Ruff (0.16.1)

[warning] Missing return type annotation for private function `other_test`

Add return type annotation: `None`

(ANN202)